### PR TITLE
fix(module:tabs): fix the pagnation padding-right when scrolling (#3539)

### PR DESCRIPTION
**Pull request by NearZXH**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #3539 

## What is the new behavior?
will have no 64 padding-right when setting `@tabs-scrolling-size` in theme.less

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by vthinkxie** [comment]
good job @NearZXH 
plz add the test and we will review your pr

**Comment by NearZXH** [comment]
@vthinkxie I have add two unit test. I add  function in nz-tabs-nav called `pxToNumber()` which is similar with the `pxToNumber()` in `components/core/util/text-measure.ts`, so i export the function, i'm not sure it's make sense.

File: components/core/util/text-measure.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ const wrapperStyle = {
   lineHeight: 'inherit'
 };
 
-function pxToNumber(value: string | null): number {
+export function pxToNumber(value: string | null): number {
```
Comment (vthinkxie) [inline components/core/util/text-measure.ts]: should use toNumber in `core/util/convert`

File: components/tabs/nz-tabs-nav.component.ts [hunk 2]
```diff
@@ -286,6 +287,19 @@ export class NzTabsNavComponent implements AfterContentChecked, AfterContentInit
     }
   }
 
+  get navContainerScrollPaddingPix(): number {
+    const navContainer = this.navContainerElement.nativeElement;
+    // tslint:disable: no-any
+    const originStyle: CSSStyleDeclaration = window.getComputedStyle
```
Comment (vthinkxie) [inline components/tabs/nz-tabs-nav.component.ts]: should make sure `this.platform.isBrowser` before use `window`

**Comment by NearZXH** [comment]
@vthinkxie I tried `toNumber()` but i think it's not suitable to such as. the originStyle.paddingLeft is a string such as '100px'，the return value is 0 through `toNumber()`; here is the original code of toNumber():
``` javascript
// in angular cdk: ~/node_modules/@angular/cdk/bundles/cdk-coercion.umd.js:Line:48
/**
 * Whether the provided value is considered a number.
 * \@docs-private
 * @param {?} value
 * @return {?}
 */
function _isNumberValue(value) {
    // parseFloat(value) handles most of the cases we're interested in (it treats null, empty string,
    // and other non-number values as NaN, where Number just uses 0) but it considers the string
    // '123hello' to be a valid number. Therefore we also check if Number(value) is NaN.
    return !isNaN(parseFloat((/** @type {?} */ (value)))) && !isNaN(Number(value));
}
```

**Comment by vthinkxie** [comment]
> @vthinkxie I tried `toNumber()` but i think it's not suitable to such situation. the originStyle.paddingLeft is a string such as '100px'，the return value is 0 through `toNumber()`; here is the original code of toNumber():
> 
> ```js
> // in angular cdk: ~/node_modules/@angular/cdk/bundles/cdk-coercion.umd.js:Line:48
> /**
>  * Whether the provided value is considered a number.
>  * \@docs-private
>  * @param {?} value
>  * @return {?}
>  */
> function _isNumberValue(value) {
>     // parseFloat(value) handles most of the cases we're interested in (it treats null, empty string,
>     // and other non-number values as NaN, where Number just uses 0) but it considers the string
>     // '123hello' to be a valid number. Therefore we also check if Number(value) is NaN.
>     return !isNaN(parseFloat((/** @type {?} */ (value)))) && !isNaN(Number(value));
> }
> ```

you are right, I have resolved the comment

File: components/tabs/nz-tabs-nav.component.ts [hunk 3]
```diff
@@ -286,6 +289,23 @@ export class NzTabsNavComponent implements AfterContentChecked, AfterContentInit
     }
   }
 
+  get navContainerScrollPaddingPix(): number {
+    if (this.platform.isBrowser) {
```
Comment (vthinkxie) [inline components/tabs/nz-tabs-nav.component.ts]: miss `PlatformModule` in the module

Comment (NearZXH) [reply components/tabs/nz-tabs-nav.component.ts]: I'm confused about whether it's necessary to add `PlatformModule` in `ngModule`, i found that there is no `PlatformModule` in `NzAffix` and `NzCarouselModule`. Can you give me more reasons to do this? `Platform` is a `Service` which is `@Injectable({providedIn: 'root'})`。 Here is the link to [PlatformModule](https://github.com/angular/components/blob/master/src/cdk/platform/platform-module.ts) and [Platform](https://github.com/angular/components/blob/master/src/cdk/platform/platform.ts) in angular.

Comment (vthinkxie) [reply components/tabs/nz-tabs-nav.component.ts]: it should be a bug in `NzAffixModule` and `NzCarouselModule` then, we have no idea what changes will happen in `PlatformModule` in the furture.

**Comment by NearZXH** [comment]
I import `PlatformModule` to `NzTabModule`. And I found there are 15 modules which are using `platform` without import `PlatformModule`, do i need to make a new pr to fix it?

**Comment by vthinkxie** [comment]
@NearZXH yes, plz help us fix it, thanks a lot!

**Pull request merged**